sdks/python/apache_beam/runners/interactive/interactive_runner.py

```diff
@@ -126,6 +127,17 @@ def apply(self, transform, pvalueish, options):
   def run_pipeline(self, pipeline, options):
     pipeline_instrument = inst.pin(pipeline, options)
 
+    # The user_pipeline analyzed might be None if the pipeline given has nothing
+    # to be cached and tracing back to the user defined pipeline is impossible.
+    # When it's None, there is no need to cache including the background
+    # caching job and no result to track since no background caching job is
+    # started at all.
+    user_pipeline = pipeline_instrument.user_pipeline
+    if user_pipeline:
+      # Should use the underlying runner and run asynchronously.
+      background_caching_job.attempt_to_run_background_caching_job(
```

Comment:
Should it be an error if this attempt fails?

Reply:
If the "fail" means returning `None` and not starting a job, then it's intended behavior. Because the logic can be invoked by previous `show` or `run` in which it might have already started a background caching job and this attempt should be NOOP.

If the attempt throws an error, the run will just error out as it should be.